### PR TITLE
Estebanreyl/data plane 08 2019

### PR DESCRIPTION
**Pull request by estebanreyl**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.  **_- Definitely tried it, never seems to leave the "still working" screen. Will update when I try again tomorrow but every other validator has succeeded.

As a note a few linting issues could not be resolved as they are intrinsic in handling compatibility with docker or OCI's own expectations for registry API's I will also be following up in email as described in https://github.com/Azure/adx-documentation-pr/wiki/Swagger-Validation-Errors-Suppression

This is a **Preview Release for the ACR runtime data-plane API specification**
This should be a fully functional release to address: https://github.com/Azure/azure-rest-api-specs/issues/6773. As an important note to this one, **my internship is coming to an end next week** and I still need to get this through before making a PR to the .NET SDK repository. I would really appreciate it if we could get this looked at quickly.

Changes:
+ Added support for most of our Blob end point API's
+ HEAD (General and For chunks), GET (General and For chunks), DELETE /v2/{name}/blobs/{digest}
+ POST (Mount from Repository), POST (Start upload) /v2/{name}/blobs/uploads/ 
+ PATCH , PUT, GET, DELETE /v2/{name}/blobs/uploads/{uuid} # Note uses some trickery to improve user experience as these require a Location link instead of parameters (As the parameters can only be obtained by parsing the Location link)
+ Added Support for OCI Manifests, Manifest Lists, and OCI Index. Models are defined for each.
+ Renamed many functions to use cleaner names.

Bugfixes:
 + Cleaned up the entire swagger file re-using parameters and definition references
 + Fixed improper use of 32 bit numbers for storing the size of certain items like layer sizes. Since some layers can exceed this size this resulted in inaccurate measurements.

All paths have been verified and the generated client has been tested as well in C#. This should ideally lead to a release of a fully functioning .NET SDK.

### CODE REVIEW

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json [hunk 1]
```diff
@@ -0,0 +1,2638 @@
...
+        "tags": [
+          "v2"
+        ],
+        "description": "Tells whether this Docker Registry instance supports Docker Registry HTTP API v2",
+        "x-ms-examples": {
+          "Check Docker Registry V2 Support": {
+            "$ref": "./examples/GetDockerRegistryV2Support.json"
+          }
+        },
+        "operationId": "CheckV2Support",
```
Comment (yungezz) [inline specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json]: operationId naming convention: Resource_Action

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json [hunk 2]
```diff
@@ -0,0 +1,2638 @@
...
+        "tags": [
+          "Manifest"
+        ],
+        "description": "Pulls the image manifest file associated with the specified name and reference. Reference may be a tag or a digest",
+        "x-ms-examples": {
+          "Get manifest": {
+            "$ref": "./examples/GetManifest.json"
+          }
+        },
+        "operationId": "GetManifest",
```
Comment (yungezz) [inline specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json]: operationId naming convention: Resource_Action

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json [hunk 3]
```diff
@@ -0,0 +1,2638 @@
...
+        "description": "Put the manifest identified by `name` and `reference` where `reference` can be a tag or digest.",
+        "x-ms-examples": {
+          "Put manifest": {
+            "$ref": "./examples/CreateManifest.json"
+          }
+        },
+        "consumes": [
+          "application/vnd.docker.distribution.manifest.v2+json"
+        ],
+        "operationId": "CreateManifest",
```
Comment (yungezz) [inline specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json]: same as above

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json [hunk 4]
```diff
@@ -0,0 +1,2638 @@
...
+                "format": "int64",
+                "description": "The length of the requested blob content."
+              },
+              "Docker-Content-Digest": {
+                "description": "Digest of the targeted content for the request.",
+                "type": "string"
+              }
+            }
+          },
+          "307": {
```
Comment (yungezz) [inline specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json]: normally not put return code other than 2xx in swagger. need ARM feedback.

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/examples/UpdateRepositoryAttributes.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "parameters": {
+    "url": "acrapi.azurecr-test.io",
+    "name": "nanoserver",
+    "value": {
+      "writeEnabled": false
+    }
+  },
+  "responses": {
+    "200": {}
```
Comment (yungezz) [inline specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/examples/UpdateRepositoryAttributes.json]: empty response body?

**Comment by estebanreyl** [comment]
Its not immediately evident to me why the pipeline tests may be failing. Looking at the results it seems like they crash on th semantic validation. Any ideas?

File: specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json [hunk 6]
```diff
@@ -0,0 +1,2638 @@
...
+          }
+        }
+      }
+    },
+    "/v2/{name}/manifests/{reference}": {
+      "get": {
+        "tags": [
+          "Manifest"
+        ],
+        "description": "Pulls the image manifest file associated with the specified name and reference. Reference may be a tag or a digest",
```
[the remark this answers is not in the document]
Comment (estebanreyl) [reply specification/containerregistry/data-plane/Microsoft.ContainerRegistry/preview/2019-08-15/containerregistry.json]: Yes, I standardized it with the others in the method as "Get the manifest identified by `name` and `reference` where `reference` can be a tag or digest."

**Comment by estebanreyl** [comment]
As a note there should be no need for an Azure REST API review board to review as described in: https://github.com/Azure/azure-rest-api-specs/pull/6629#issuecomment-515623916 . This follows as the API is already in production.

**Comment by estebanreyl** [comment]
I don't want to be pushy but I am unfortunately running out of time here any updates on this front?

**Comment by yungezz** [comment]
Hi @johanste , could you pls help to review the PR? thanks.

**Review by yungezz — APPROVED** [review]

**Pull request merged**